swarm/src/behaviour.rs

```diff
@@ -26,6 +26,24 @@ use std::{error, task::Context, task::Poll};
 ///
 /// This trait has been designed to be composable. Multiple implementations can be combined into
 /// one that handles all the behaviours at once.
+///
+/// # Deriving `NetworkBehaviour`
+///
+/// Crate users should implement this trait by using the the `#[derive(NetworkBehaviour)]`
+/// proc macro re-exported by `libp2p` crate. The macro generates a delegating `trait`
+/// implementation for the `struct`, which delegates method calls to all trait members. Any events
+/// generated by struct members are delegated to
+/// [`NetworkBehaviourEventProcess`](trait.NetworkBehaviourEventProcess.html) implementations which
```

Comment:
```suggestion
/// [`NetworkBehaviourEventProcess`] implementations which
```
We already use [intra-rustdoc-links](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md).

Reply:
Oh ok. Initially I thought these were on stable but apparently not and then added the links manually. I'll get rid of these and combine your other suggestions.